src/main/java/seedu/address/model/EntityList/ParticipantList.java

```diff
@@ -0,0 +1,114 @@
+package seedu.address.model.EntityList;
+
+import java.util.ArrayList;
+import java.util.List;
+import seedu.address.model.Entity.Entity;
+import seedu.address.model.Entity.ID;
+import seedu.address.model.Entity.Participant;
+import seedu.address.model.Entity.PrefixType;
+
+public class ParticipantList implements EntityListInterface {
+    private List<Participant> participants;
+    private int nextIDSuffix;
+
+    /**
+     * Constructor.
+     */
+    public ParticipantList() {
+       this.participants = new ArrayList<>();
+       this.nextIDSuffix = 1;
+    }
+
+    /**
+     * Gets participant by id.
+     *
+     * @param id
+     * @return
+     */
+    @Override
+    public Participant get(ID id) {
+        return new Participant("name", "email", "999", this.generateID());
+    }
+
+    /**
+     * Updates participant by id.
+     *
+     * @param participant
+     * @throws Exception if error while updating.
+     */
+    @Override
+    public void update(Entity participant) throws Exception {
+        // TODO: Find by ID then replace.
+        participants.add((Participant) participant);
+    }
+
+    /**
+     * Adds participant to the list.
+     *
+     * @param participant
+     * @throws Exception if there was an error while adding.
+     */
+    @Override
+    public void add(Entity participant) throws Exception {
+        // TODO: Find by ID then replace.
+        participants.add((Participant) participant);
+    }
+
+    /**
+     * Deletes participant by ID.
+     *
+     * @param id
+     * @throws Exception if error while deleting.
+     */
+    @Override
+    public void delete(ID id) throws Exception {
+        for (Participant p: this.participants) {
+            if (p.getId() == id) {
+                this.participants.remove(p);
+                return;
+            }
+        }
+    }
+
+    /**
+     * List the participants.
+     *
+     * @return List<Participant>
+     */
+    @Override
+    public List<? extends Entity> list() {
+        return this.participants;
+    }
+
+    /**
+     * Checks if a given ID exists.
+     *
+     * @param id
+     * @return boolean
+     */
+    @Override
+    public boolean isContain(ID id) {
+        for (Participant p: this.participants) {
+            if (p.getId() == id) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Generates the ID for the object.
+     *
+     * @return ID
+     */
+    @Override
+    public ID generateID() {
```

Comment:
Do we need an ID for `XLists` too? What will it be used for?

Reply:
This would be for generating the ID of the entity in the list when you create a new Entity. My reason for adding this in was because I felt that the IDs should be generated by the various lists instead of the controller. So before creating the new Entity, the respective controller will call generateID to get the ID of the new object

Reply:
Static works, though I think that generateId should be based in the list because it seems to be the responsibility of the container to generateID, as opposed to the list.

Also thinking of switching the IDs to [this](https://stackoverflow.com/questions/1389736/how-do-i-create-a-unique-id-in-java) as opposed to numbers as I can see some issues potentially surfacing with simple increments

Reply:
Yea I think generateId should be based in XLists. But if we are going to require CLI to take IDs as input, wouldn't this be too complicated?

Reply:
Yep it would. hmm if that happens then we'll overload get with another public method that takes in like name? wdyt?

Reply:
Yea we should do that. I think at some stage, we should just remove adding, deleting etc by id of the entity. It'll be too complicated from the user's perspective.